marklogic-client-api/src/test/java/com/marklogic/client/test/rows/RowManagerValidateDocTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2022 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.client.test.rows;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.WriteBatcher;
+import com.marklogic.client.document.DocumentManager;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.impl.DocumentWriteOperationImpl;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.query.DeleteQueryDefinition;
+import com.marklogic.client.query.QueryManager;
+import com.marklogic.client.row.RowManager;
+import com.marklogic.client.row.RowRecord;
+import com.marklogic.client.test.Common;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static com.marklogic.client.io.Format.JSON;
+import static com.marklogic.client.io.Format.XML;
+import static org.junit.Assert.assertTrue;
+
+public class RowManagerValidateDocTest {
+    private final static String DIRECTORY = "/validateDocTest/";
+    private static Set<String> set = new HashSet<>();
+    protected ObjectMapper mapper = new ObjectMapper();
+    String validateDocCollection = "RowManagerValidateDocTest";
+    static DataMovementManager dataMovementManager;
+    static RowManager rowManager;
+    static PlanBuilder op;
+
+    @BeforeClass
+    public static void setUp(){
+        Common.connect();
+        dataMovementManager = Common.client.newDataMovementManager();
+        rowManager = Common.client.newRowManager();
+        op = rowManager.newPlanBuilder();
+    }
+
+
+    @Test
+    public void validateDocWithxmlSchema() {
```

Comment:
For each type of schema validation, we need both a positive and negative test to ensure that the validation is working correctly. Additionally, we need a separate test for each "mode" to ensure that it's working correctly as well.